docs/customization/controller.rst

```diff
@@ -0,0 +1,142 @@
+
+            return $this->render('SyliusWebBundle:Frontend/Homepage:_advantages.html.twig', ['bestsellers' => $bestsellers]);
+        }
+    }
+
+2. In order to use your controller and its actions you need to configure it in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    sylius_product:
+        resources:
+            product:
+                classes:
+                    controller: AppBundle\Controller\ProductController
+
+How to customize a Frontend Controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to add the rendering of bestsellers on your Homepage - to do this you should modify the ``mainAction`` of the ``HomepageController``.
+You should already have a method on the ``ProductRepository`` (you can see such an example :doc:`here </customization/repository>` that you can call in the controller.
+
+1. Create a new Controller class under the ``AppBundle/Controller/Frontend`` namespace.
+
+Run ``$ php app/console debug:container sylius.controller.frontend.homepage``.
+Your class needs to be extending this base class.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Controller\Frontend;
+
+    use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+    use Sylius\Component\Resource\Repository\RepositoryInterface;
+    use Sylius\Bundle\WebBundle\Controller\Frontend\HomepageController as BaseHomepageController;
+    use Symfony\Component\HttpFoundation\Response;
+
+    /**
+     * @author Name Surname <name.surname@test.com>
+     */
+    class HomepageController extends BaseHomepageController
+    {
+        /**
+         * @return Response
+         */
+        public function mainAction()
+        {
+            /** @var RepositoryInterface $customEntityRepository */
+            $customEntityRepository = $this->get('app.repository.custom_entity');
+            /** @var ProductRepositoryInterface $productRepository */
+            $productRepository = $this->get('sylius.repository.product');
+
+            $customEntities = $customEntityRepository->findBy(['criteria' => true]);
+            $bestsellers = $productRepository->findBySold(4);
+
+            return $this->render(
+                'SyliusWebBundle:Frontend/Homepage:main.html.twig',
+                [
+                    'customEntities' => $customEntities,
+                    'bestsellers' => $bestsellers,
+                ]
+            );
+        }
+    }
+
+2. The next thing you have to do is to override the ``sylius.controller.frontend.homepage.class`` parameter in ``AppBundle/Resources/config/services.yml``.
+
+.. code-block:: yaml
+
+    parameters:
+        sylius.controller.frontend.homepage.class: AppBundle\Controller\Frontend\HomepageController
+
+From now on your ``mainAction`` of the ``HomepageController`` will be rendering bestsellers, that will be available
+in the ``SyliusWebBundle:Frontend/Homepage:main.html.twig`` view under the ``bestsellers`` variable.
```

Comment:
This customization is also not a perfect example, because you should use a partial to render that list. That allows for easy ESI/caching and is simpler. So we should think about a bit different use case.